internal/parser/parser.go

```diff
@@ -25,26 +26,32 @@ import (
 	configurationv1beta1 "github.com/kong/kubernetes-ingress-controller/pkg/apis/configuration/v1beta1"
 )
 
-func parseAll(log logrus.FieldLogger, s store.Storer) ingressRules {
+func parseAll(log logrus.FieldLogger, s store.Storer, promMetrics *util.ControllerFunctionalPrometheusMetrics) ingressRules {
```

Comment:
For metrics, looking at individual stages of `parseAll` doesn't seem to give the user a useful nor clear signal as to whether "KIC was able to produce a valid kong configuration" or not.

Instead of incrementing `ParseCounter` on each of the steps within `parseAll`, please increment the counter only once: after a call to `parser.Build`.

Reply:
addressed https://github.com/Kong/kubernetes-ingress-controller/pull/1622/commits/84df128254b7b56653f176a87b6d3efa470dd2c8